src/movies.js

```diff
@@ -1,16 +1,115 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+  let directors = movies.map((movie) => movie.director);
+  return directors;
+
+  /*let directors = movies.map (function(movie){
+    return movie.director;
+  });
+  return directors; */
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movieArr) {
+  const spielbergMovies = movieArr.filter(function (movie) {
+    return (
+      movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+    );
+  });
+  return spielbergMovies.length;
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movie) {
+  if (movie.length === 0) {
+    return 0;
+  }
+  const avg = movie.reduce(function (acc, value) {
+    if (!value.rate) {
+      return acc;
+    }
+    return acc + value.rate;
+  }, 0);
+  //   let rates = movie.map(function (element) {
+  //     return element.rate;
+  //   });
+  //   let onlyrate = rates.filter(function (element) {
+  //     return element;
+  //   });
+  //   let average = onlyrate.reduce(function (acc, value) {
+  //     return acc + value;
+  //   });
+  let finalavg = avg / movie.length;
+  return Math.round(finalavg * 100) / 100;
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movie) {
+  let drama = movie.filter(function (element) {
+    if (element.genre.includes("Drama")) {
+      return element.rate;
+    }
+  });
+  let rates = drama.map(function (element) {
+    return element.rate;
+  });
+  if (rates.length === 0) {
+    return 0;
+  }
+  let avg = rates.reduce(function (acc, num) {
+    return acc + num;
+  });
+  let result = avg / rates.length;
+  return Math.round(result * 100) / 100;
+}
```

Comment:
Could also have reused the previous ratesAverage() function to calculate the average here! 😄